tests/prague/eip7692_eof_v1/eip7069_extcall/test_calldata.py

```diff
@@ -0,0 +1,288 @@
+"""
+abstract: Tests [EIP-7069: Revamped CALL instructions](https://eips.ethereum.org/EIPS/eip-7069)
+    Tests for the RETURNDATALOAD instriction
+"""  # noqa: E501
+
+import pytest
+
+from ethereum_test_tools import Account, Alloc, Environment, StateTestFiller, Transaction
+from ethereum_test_tools.eof.v1 import Container, Section
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+
+from .. import EOF_FORK_NAME
+from . import REFERENCE_SPEC_GIT_PATH, REFERENCE_SPEC_VERSION
+from .helpers import (
+    slot_call_status,
+    slot_calldata_1,
+    slot_calldata_2,
+    slot_code_worked,
+    slot_delegate_code_worked,
+    value_code_worked,
+    value_exceptional_abort_canary,
+)
+from .spec import EXTCALL_SUCCESS
+
+REFERENCE_SPEC_GIT_PATH = REFERENCE_SPEC_GIT_PATH
+REFERENCE_SPEC_VERSION = REFERENCE_SPEC_VERSION
+
+pytestmark = pytest.mark.valid_from(EOF_FORK_NAME)
+
+
+def right_pad_32(v: bytes) -> bytes:
+    """Takes bytes and returns a 32 byte version right padded with zeros"""
+    return v + b"\0" * (32 - len(v))
+
+
+@pytest.mark.parametrize("value", [0, 1])
+@pytest.mark.parametrize(
+    "memory",
+    [
+        b"",
+        b"1234567890abcdef",
+        b"1234567890abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ=-",
+        b"1234567890abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ=-" * 4,
+    ],
+    ids=lambda x: "size_%d" % len(x),
+)
+@pytest.mark.parametrize("offset", [0, 8, 24, 80])
+@pytest.mark.parametrize("length", [0, 8, 32, 48])
+def test_extcalls_inputdata(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    value: int,
+    memory: bytes,
+    offset: int,
+    length: int,
+):
+    """
+    Tests call data into EXT*CALL including multiple offset conditions.
+
+    Caller pushes data into memory, then calls the target.  Target writes 64 bytes of call data
+    to storage and a success byte.
+    """
+    env = Environment()
+
+    sender = pre.fund_eoa(10**18)
+
+    address_returner = pre.deploy_contract(
+        Container(
+            sections=[
+                Section.Code(
+                    code=Op.SSTORE(slot_calldata_1, Op.CALLDATALOAD(0))
+                    + Op.SSTORE(slot_calldata_2, Op.CALLDATALOAD(32))
```

Comment:
This might actually be a prompt to a different test, but it would be interesting to insert an Op.EXTCALL before this and perform a similar verification inside that call too, to check that the CALLDATA doesn't spill into the next call level, and also does not affect the CALLDATA in this level after returning.
```suggestion
                    code=Op.EXTCALL(address_subcall, 0, 0, 0)
                    + Op.SSTORE(slot_calldata_1, Op.CALLDATALOAD(0))
                    + Op.SSTORE(slot_calldata_2, Op.CALLDATALOAD(32))
```
`address_subcall` could contain `Op.SSTORE(slot_calldata_1, Op.CALLDATALOAD(0))  + ...` too.

Reply:
Different test.  I'll give it a quick swag.

Reply:
added